src/main/java/com/angelbroking/smartapi/http/SmartAPIRequestHandler.java

```diff
@@ -1,420 +1,379 @@
-	 * Creates a PUT request.
-	 * 
-	 * @param url         is the endpoint to which request has to be done.
-	 * @param apiKey      is the api key of the Smart API Connect app.
-	 * @param accessToken is the access token obtained after successful login
-	 *                    process.
-	 * @param params      is the map of data that has to be sent in the body.
-	 */
-	public Request createPutRequest(String url, Map<String, Object> params, String apiKey, String accessToken) {
-		FormBody.Builder builder = new FormBody.Builder();
-		for (Map.Entry<String, Object> entry : params.entrySet()) {
-			builder.add(entry.getKey(), entry.getValue().toString());
-		}
-		RequestBody requestBody = builder.build();
-		Request request = new Request.Builder().url(url).put(requestBody).header("User-Agent", USER_AGENT)
-				.header("X-Smart API-Version", "3").header("Authorization", "token " + apiKey + ":" + accessToken)
-				.build();
-		return request;
-	}
-
-	/**
-	 * Creates a DELETE request.
-	 * 
-	 * @param url         is the endpoint to which request has to be done.
-	 * @param apiKey      is the api key of the Smart API Connect app.
-	 * @param accessToken is the access token obtained after successful login
-	 *                    process.
-	 * @param params      is the map of data that has to be sent in the query
-	 *                    params.
-	 */
-	public Request createDeleteRequest(String url, Map<String, Object> params, String apiKey, String accessToken) {
-		HttpUrl.Builder httpBuilder = HttpUrl.parse(url).newBuilder();
-		for (Map.Entry<String, Object> entry : params.entrySet()) {
-			httpBuilder.addQueryParameter(entry.getKey(), entry.getValue().toString());
-		}
-
-		Request request = new Request.Builder().url(httpBuilder.build()).delete().header("User-Agent", USER_AGENT)
-				.header("X-Smart API-Version", "3").header("Authorization", "token " + apiKey + ":" + accessToken)
-				.build();
-		return request;
-	}
+    private static final Logger logger = LoggerFactory.getLogger(SmartAPIRequestHandler.class);
+    private static final String USER_AGENT = "javasmartapiconnect/3.0.0";
+    private final OkHttpClient client;
+    JSONObject apiheader = apiHeaders();
+
+    /**
+     * Initialize request handler.
+     *
+     * @param proxy to be set for making requests.
+     */
+    public SmartAPIRequestHandler(Proxy proxy) {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.connectTimeout(10000, TimeUnit.MILLISECONDS);
+        if (proxy != null) {
+            builder.proxy(proxy);
+        }
+
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
+        if (SmartConnect.enableLogging) {
+            client = builder.addInterceptor(logging).build();
+        } else {
+            client = builder.build();
+        }
+    }
+
+    public JSONObject apiHeaders() {
+
+        try {
+            JSONObject headers = new JSONObject();

```

Comment:
Why can't we use HttpHeaders here instead of JSONObject?

Reply:
have used POJO class ApiHeaders instead of JSONObject